terraform.sh

```diff
@@ -72,6 +72,14 @@ function end_execution() {
   # Delete trap handler to avoid recursion
   trap - HUP QUIT PIPE INT TERM EXIT
 
+  if [ -n "$TF_PID" ] && kill -0 "$TF_PID" &>/dev/null; then
+    echo "$(date) Sending SIGTERM to terraform process $TF_PID."
+    kill -SIGTERM $TF_PID
+    echo "$(date) Waiting for terraform process $TF_PID to complete..."
+    wait $TF_PID
```

Comment:
I spotted an issue here - wait exits when the exit code != 0 and this prevents the state ConfigMap to be updated. When may have a case in which terraform apply is running, SIGTERM is received, we reach this wait and terraform process fail with conflicts EntityAlreadyExists and the state ConfigMap is not updated.

/hold

Reply:
Thanks for catching it!